CMake/External_OpenCV.cmake

```diff
@@ -21,6 +24,23 @@ else()
   set(fletch_ENABLE_OpenCV_FFmpeg CACHE INTERNAL FALSE)
 endif()
 
+# Allow OpenCV's VTK option to be explicitly turned off
+if(fletch_ENABLE_VTK)
+  # By default turn opencv-VTK support off, even when building with VTK
+  option(fletch_ENABLE_OpenCV_VTK "Build OpenCV with VTK support" FALSE )
+  mark_as_advanced(fletch_ENABLE_OpenCV_VTK)
+else()
+  set(fletch_ENABLE_OpenCV_VTK CACHE INTERNAL FALSE)
```

Comment:
Yes, there is defiantly a problem with this. It exists for the other flags like `fletch_ENABLE_OpenCV_CUDA` and `fletch_ENABLE_OpenCV_FFmpeg` as well. I originally had rewrote all of them to attempt to fix the issue, but I thought my solution was too complex, so I just removed it and followed the pattern of the other variables. 

For reference my solution was to make a special `advanced_option` function that was only visible if a condition was met. 

```cmake
function(advanced_option varname docstring default is_relevant)
  # Allows advanced options to be shown in the gui only when appropriate
  # Once an variable is set as CACHE INTERAL, it seems the only way to make it
  # visible in the gui again is to unset it.
  if (DEFINED ${varname})
    # Remember the previous value of the variable
    set(value ${${varname}})
    unset(${varname} CACHE)
  else()
    # Use the default on the first call or if the var has been externally unset
    set(value ${default})
  endif()

  if (${is_relevant})
    # Show the last known value of the variable, but only in advanced mode
    option(${varname} ${docstring} ${value})
    mark_as_advanced(${varname})
  else()
    # Keep the variable around in internal cache, but dont show it even in
    # advanced mode.
    set(${varname} ${value} CACHE INTERNAL ${docstring})
  endif()
endfunction()
```